nbbuild/antsrc/org/netbeans/nbbuild/CheckLinks.java

```diff
@@ -633,8 +645,33 @@ final Boolean isOk (URI u) throws BuildException {
                 throw new BuildException ("Each filter must have pattern attribute");
             }
             
-            if (pattern.matcher (u.toString ()).matches ()) {
-                log ("Matched " + u + " accepted: " + accept, org.apache.tools.ant.Project.MSG_VERBOSE);
+            if (pattern.matcher(u.toString()).matches()) {
+                log("Matched " + u + " accepted: " + accept, org.apache.tools.ant.Project.MSG_VERBOSE);
+                if (externallinksdump != null) {
+                    try {
+                        String dummyName;
+                        if (accept) {
+                            dummyName = "acceptednetbeans.txt";
+                        } else {
+                            dummyName = "rejectednetbeans.txt";
+                        }
+                        Path dumppath = externallinksdump.toPath().resolve(dummyName);
+                        if (Files.notExists(dumppath)) {
+                            Files.createDirectories(dumppath.getParent());
+                            Files.createFile(dumppath);
+                        }
+                        List<String> existingEntries = Files.readAllLines(externallinksdump.toPath().resolve(dummyName));
+                        Set<String> sortedEntries = new TreeSet<>(existingEntries);
+                        sortedEntries.add(u.toString());
+                        StringBuilder sb = new StringBuilder();
+                        for (String sortedEntry : sortedEntries) {
+                            sb.append(sortedEntry).append(System.lineSeparator());
+                        }
+                        Files.write(dumppath, sb.toString().getBytes(), StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
```

Comment:
Wouldn't 
Files.write(dumbpath, sortedEntries, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
be more efficient?

Reply:
+1